mapper-processor/src/main/java/com/datastax/oss/driver/internal/mapper/processor/dao/DaoImplementationGenerator.java

```diff
@@ -70,13 +79,45 @@
   private final List<GeneratedQueryProvider> queryProviders = new ArrayList<>();
   private final NullSavingStrategyValidation nullSavingStrategyValidation;
 
+  // tracks interface type variable mappings as child interfaces discover them.
+  private final Map<TypeMirror, Map<Name, TypeElement>> typeMappingsForInterface =
+      Maps.newHashMap();
+
+  private final Set<TypeMirror> interfaces;
+  private final Map<Class<? extends Annotation>, Annotation> annotations;
+
+  private static final Set<Class<? extends Annotation>> ANNOTATIONS =
+      ImmutableSet.of(DefaultNullSavingStrategy.class);
+
   public DaoImplementationGenerator(TypeElement interfaceElement, ProcessorContext context) {
     super(context);
     this.interfaceElement = interfaceElement;
+    this.interfaces = HierarchyScanner.resolveTypeHierarchy(interfaceElement, context);
+    this.annotations = scanAnnotations();
     implementationName = GeneratedNames.daoImplementation(interfaceElement);
     nullSavingStrategyValidation = new NullSavingStrategyValidation(context);
   }
 
+  private Map<Class<? extends Annotation>, Annotation> scanAnnotations() {
+    Map<Class<? extends Annotation>, Annotation> annotations = Maps.newHashMap();
+    for (TypeMirror mirror : interfaces) {
+      Element element = context.getTypeUtils().asElement(mirror);
+      for (Class<? extends Annotation> annotationClass : ANNOTATIONS) {
+        Annotation annotation = element.getAnnotation(annotationClass);
+        if (annotation != null) {
+          // don't replace annotations from lower levels.
+          annotations.putIfAbsent(annotationClass, annotation);
+        }
+      }
+    }
+    return ImmutableMap.copyOf(annotations);
+  }
+
+  @SuppressWarnings("unchecked")
+  private <A> Optional<A> getAnnotation(Class<A> annotationClass) {
+    return Optional.ofNullable((A) annotations.get(annotationClass));
```

Comment:
```suggestion
    return Optional.ofNullable(annotationClass.cast(annotations.get(annotationClass)));
```
Essentially the same, but it eliminates the need for the `@SuppressWarnings` on the method.